core/src/main/scala/za/co/absa/spline/harvester/builder/WindowNodeBuilder.scala

```diff
@@ -16,19 +16,31 @@
 
 package za.co.absa.spline.harvester.builder
 
-import org.apache.spark.sql.catalyst.expressions.{Alias, Attribute, Expression}
+import org.apache.spark.sql.catalyst.expressions.{Alias, Attribute, Expression, NamedExpression}
 import org.apache.spark.sql.catalyst.plans.logical.Window
+import za.co.absa.commons.reflect.ReflectionUtils
 import za.co.absa.spline.harvester.ComponentCreatorFactory
 import za.co.absa.spline.harvester.postprocessing.PostProcessor
 
+import scala.util.Try
+
 class WindowNodeBuilder
   (override val operation: Window)
   (override val componentCreatorFactory: ComponentCreatorFactory, postProcessor: PostProcessor)
   extends GenericNodeBuilder(operation)(componentCreatorFactory, postProcessor) {
 
   override def resolveAttributeChild(attribute: Attribute): Option[Expression] = {
-    operation.windowExpressions
+    extractNamedExpressions(operation)
       .find(_.exprId == attribute.exprId)
       .map(_.asInstanceOf[Alias])
   }
+
+  private def extractNamedExpressions(operation: Window): Seq[NamedExpression] = {
```

Comment:
I would add a comment explaining why just `.name` doesn't suffice, with the reference to this issue.